src/pyodide/internal/workers.py

```diff
@@ -457,3 +466,137 @@ def name(self) -> str:
     @property
     def last_modified(self) -> int:
         return self._js_blob.last_modified
+
+
+class Request:
+    def __init__(self, input: "Request | str", **other_options: Unpack[FetchKwargs]):
+        if "method" in other_options and isinstance(
+            other_options["method"], HTTPMethod
+        ):
+            other_options["method"] = other_options["method"].value
+
+        if "headers" in other_options:
+            other_options["headers"] = _to_js_headers(other_options["headers"])
+        self._js_request = js.Request.new(
+            input._js_request if isinstance(input, Request) else input, **other_options
+        )
+
+    @property
+    def js_object(self) -> "js.Request":
+        return self._js_request
+
+    # TODO: expose `body` as a native Python stream in the future, follow how we define `Response`
+    @property
+    def body(self) -> "js.ReadableStream":
+        return self.js_object.body
+
+    @property
+    def body_used(self) -> bool:
+        return self.js_object.bodyUsed
+
+    @property
+    def cache(self) -> str:
+        return self.js_object.cache
+
+    @property
+    def credentials(self) -> str:
+        return self.js_object.credentials
+
+    @property
+    def destination(self) -> str:
+        return self.js_object.destination
+
+    @property
+    def headers(self) -> http.client.HTTPMessage:
+        result = http.client.HTTPMessage()
```

Comment:
Is there a reason to use `HTTPMessage` here instead of a dict? I'm pretty sure that the JavaScript `Headers` object will never give us multiple copies of the same key, instead it combines them with a comma. Or perhaps we should split the values on commas and turn them back into multiple entries?

Reply:
The main reason is because it supports multiple header values with the same key.

But good catch, I thought the iterator in JS would yield all the combinations without the commas but that doesn't seem to be the case.